wok/p5/sketch.js

```diff
@@ -1,84 +1,132 @@
 import "p5";
+import Grid from "../models/Grid";
 
-import grid from "./grid";
+let xInput,
+  yInput,
+  zInput,
+  setSizebutton,
+  titleText,
+  xText,
+  yText,
+  zText,
+  grid,
+  tiles = [],
+  food = [];
+
+const tileSize = 80;
+const gap = 0;
 
 function setup() {
   frameRate(30);
 
   const canvas = createCanvas(1280, 960, WEBGL);
   canvas.parent("canvas-container");
+
+  createCubeConfigForm(10, 0);
+}
+
+function createCubeConfigForm(posX, posY) {
+  titleText = createElement("h2", "Cube Config");
+  titleText.position(posX + 20, posY + 5);
+
+  xText = createElement("p", "X:");
+  xText.position(posX + 5, posY + 45);
+
+  yText = createElement("p", "Y:");
+  yText.position(posX + 5, posY + 75);
+
+  zText = createElement("p", "Z:");
+  zText.position(posX + 5, posY + 105);
+
+  xInput = createInput();
+  xInput.position(posX + 25, posY + 60);
+
+  yInput = createInput();
+  yInput.position(posX + 25, posY + 90);
+
+  zInput = createInput();
+  zInput.position(posX + 25, posY + 120);
+
+  setSizebutton = createButton("Set Size");
+  setSizebutton.position(posX + 75, posY + 150);
+  setSizebutton.mousePressed(onSetCubeSize);
+}
+function onSetCubeSize() {
+  if (xInput.value() != "" && yInput.value() != "" && zInput.value() != "") {
+    grid = new Grid(
+      parseInt(xInput.value()),
+      parseInt(yInput.value()),
+      parseInt(zInput.value())
+    );
+    for (let key of tiles) {
+      const c = key.split("|").map(i => +i);
+      grid.get(grid.toIndex(...c)).type = map.tiles[key];
+    }
+    for (let key in food) {
```

Comment:
What is this food here?

Reply:
"food" was already used in the code. We reused the existing code. Miran said, "food is part of the maze code, it's used as a reward for path finder".

Reply:
There's no need for food here. The path will be just random walk.

Reply:
Removed food,

Reply:
![image](https://user-images.githubusercontent.com/29366282/72315314-5c618a00-36b8-11ea-9ee3-3ea77bb0dfaf.png)